src/components/buyers/list/BuyerList.tsx

```diff
@@ -0,0 +1,232 @@
+    header: "Buyer ID",
+    accessor: "ID",
+    cell: ({row, value}) => (
+      <Link passHref href={"/buyers/" + row.original.ID}>
+        <Text as="a" noOfLines={2} title={value}>
+          {value}
+        </Text>
+      </Link>
+    )
+  }
+
+  const NameColumn: DataTableColumn<IBuyer> = {
+    header: "NAME",
+    accessor: "Name",
+    cell: ({row, value}) => (
+      <Link passHref href={"/buyers/" + row.original.ID}>
+        <Text as="a" noOfLines={2} title={value}>
+          {value}
+        </Text>
+      </Link>
+    )
+  }
+
+  const DefaultCatalogIDColumn: DataTableColumn<IBuyer> = {
+    header: "DEFAULT CATALOG ID",
+    accessor: "DefaultCatalogID"
+  }
+
+  const StatusColumn: DataTableColumn<IBuyer> = {
+    header: "STATUS",
+    accessor: "Active",
+    cell: ({row}) => (
+      <>
+        <Icon
+          as={row.original.Active === true ? MdCheck : IoMdClose}
+          color={row.original.Active === true ? "green.400" : "red.400"}
+          w="20px"
+          h="20px"
+        />
+        <Text>{row.original.Active ? "Active" : "Non active"}</Text>
+      </>
+    )
+  }
+
+  const CreatedDateColumn: DataTableColumn<IBuyer> = {
+    header: "CREATED DATE",
+    accessor: "DateCreated",
+    cell: ({value}) => dateHelper.formatDate(value)
+  }
+
+  const UserGroupColumn: DataTableColumn<IBuyer> = {
+    header: "USER GROUPS / USERS",
+    cell: ({row}) => (
+      <ButtonGroup>
+        <Button onClick={() => router.push(`/buyers/${row.original.ID}/usergroups`)} variant="outline">
+          User Groups ({buyersMeta[row.original.ID]["userGroupsCount"]})
+        </Button>
+        <Button onClick={() => router.push(`/buyers/${row.original.ID}/users`)} variant="outline">
+          Users ({buyersMeta[row.original.ID]["usersCount"]})
+        </Button>
+      </ButtonGroup>
+    )
+  }
+
+  const CatalogColumn: DataTableColumn<IBuyer> = {
+    header: "CATALOGS",
+    cell: ({row}) => (
+      <Link href={`/buyers/${row.original.ID}/catalogs`}>
+        <Button variant="outline">Catalogs ({buyersMeta[row.original.ID]["catalogsCount"]})</Button>
+      </Link>
+    )
+  }
+
+  const ActionsColumn: DataTableColumn<IBuyer> = {
+    header: "ACTIONS",
+    cell: ({row}) => (
+      <ButtonGroup>
+        <Button variant="outline" onClick={() => router.push(`/buyers/${row.original.ID}/`)}>
+          Edit
+        </Button>
+        <Button variant="outline" onClick={() => deleteBuyer(row.original.ID)}>
+          Delete
+        </Button>
+      </ButtonGroup>
+    )
+  }
+
+  const BuyerTableOptions: ListViewTableOptions<IBuyer> = {
+    responsive: {
+      base: [IdColumn, NameColumn],
+      md: [IdColumn, NameColumn],
+      lg: [IdColumn, NameColumn],
+      xl: [
+        IdColumn,
+        NameColumn,
+        DefaultCatalogIDColumn,
+        StatusColumn,
+        CreatedDateColumn,
+        UserGroupColumn,
+        CatalogColumn,
+        ActionsColumn
+      ]
+    }
+  }
+
+  return (
+    showList && (
```

Comment:
The ListView should be smart enough to handle something like skeleton components for the initial load (it does not use suspense at the moment); however, I see no reason to wait to show the entire list view until after the data is loaded. You could show the toolbar, and the loading indicator should naturally display on the sub table/grid.

Reply:
I guess what I'm saying here is, remove the showList conditional and state - it seems unnecessary. And if you are getting errors for some reason without this conditional, let's talk about it.

Reply:
@robertsoniv While cleaning this up I remembered why I was doing it.  It is related to the UserGroupColumn that needs the buyersMeta data before it can render, otherwise an error is thrown.